src/components/TextInput/index.js

```diff
@@ -22,6 +23,7 @@ class TextInput extends React.Component {
             <BaseTextInput
                 // eslint-disable-next-line react/jsx-props-no-spreading
                 {...this.props}
+                shouldWaitWidthCalculation={BrowserUtils.isMobile()}
```

Comment:
As @parasharrajat posted, I agree it would be good to investigate this further before just disabling it for desktop. If we don't understand the cause, we may be creating a bug.